README.md

```diff
@@ -1,3 +1,92 @@
-# Renovate Log Analyzer
+# Renovate Log Analyzer (used as part of the [MintMaker](https://github.com/konflux-ci/mintmaker) service)
 
-A tool for analyzing Renovate logs to detect typical issues.
+This repository contains a Go implementation for analyzing Renovate logs. The implementation provides level-based error (and fatal) extraction.
+
+Another part of this repo is the KITE client, which in the event of a failure in Renovate sends the extracted errors to the [KITE API](https://github.com/konflux-ci/kite) to be displayed on an Issue dashboard in Konflux UI.
+
+This service is meant to run as last step of `tekton pipeline` created by the [MintMaker controller](https://github.com/konflux-ci/mintmaker).
+
+## Log analyzer
+
+- **`models.go`**: Data models (`LogEntry` and `PodDetails`)
+- **`log_reader.go`**: Log processing logic for extracting logs from a `json` file and parsing them into `Go` object as well as the analyzing logic
+
+## Log Levels
+
+Following [Renovate documentation](https://docs.renovatebot.com/troubleshooting/):
+
+- **TRACE**: 10
+- **DEBUG**: 20
+- **INFO**: 30
+- **WARN**: 40
+- **ERROR**: 50
+- **FATAL**: 60
+
+The log-analyzer looks for errors with level 50 (ERROR) and 60 (FATAL) and extracts the most useful information from them by looking through all the possible fields in their structure. It aggregates duplicate errors and formats them into a summary message.
+
+## KITE client
```

Comment:
Kite isn't an acronym, so I suggest keeping the naming consistent by using kite/Kite.